test/e2e/healthcheck_test.go

```diff
@@ -47,9 +60,201 @@ var _ = Describe("Health Check Test", Serial, Labels{"health_checks"}, func() {
+			Eventually(func(g Gomega) {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsRecord), dnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dnsRecord.Status.Endpoints).To(HaveLen(5))
+				g.Expect(dnsRecord.Status.Endpoints).To(ContainElement(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"DNSName":    Equal("14byhk-2k52h1.klb." + testHostname),
+						"Targets":    ConsistOf("172.32.200.1"),
+						"RecordType": Equal("A"),
+					})),
+				))
+				g.Expect(dnsRecord.Status.Conditions).To(ContainElements(
+					MatchFields(IgnoreExtras, Fields{
+						"Type":    Equal(string(v1alpha1.ConditionTypeReady)),
+						"Status":  Equal(metav1.ConditionFalse),
+						"Reason":  Equal(string(v1alpha1.ConditionReasonUnhealthy)),
+						"Message": Equal("None of the healthchecks succeeded"),
+					}),
+					MatchFields(IgnoreExtras, Fields{
+						"Type":    Equal(string(v1alpha1.ConditionTypeHealthy)),
+						"Status":  Equal(metav1.ConditionFalse),
+						"Reason":  Equal(string(v1alpha1.ConditionReasonUnhealthy)),
+						"Message": And(ContainSubstring("Not healthy addresses:"), ContainSubstring("172.32.200.1"), ContainSubstring("172.32.200.2")),
+					}),
+				))
+			}, TestTimeoutLong, time.Second, ctx).Should(Succeed())
+
+			By("removing healthchecks")
+			Eventually(func(g Gomega) {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsRecord), dnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				dnsRecord.Spec.HealthCheck = nil
+
+				g.Expect(k8sClient.Update(ctx, dnsRecord)).To(Succeed())
+			}, TestTimeoutMedium, time.Second, ctx).Should(Succeed())
+
+			By("ensure EPs are published")
+			Eventually(func(g Gomega) {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsRecord), dnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dnsRecord.Status.Endpoints).To(HaveLen(5))
+				g.Expect(dnsRecord.Status.Endpoints).To(ContainElement(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"DNSName":    Equal("14byhk-2k52h1.klb." + testHostname),
+						"Targets":    ConsistOf("172.32.200.1", "172.32.200.2"),
+						"RecordType": Equal("A"),
+					})),
+				))
+				g.Expect(dnsRecord.Status.Conditions).To(ContainElements(
+					MatchFields(IgnoreExtras, Fields{
+						"Type":    Equal(string(v1alpha1.ConditionTypeReady)),
+						"Status":  Equal(metav1.ConditionTrue),
+						"Reason":  Equal(string(v1alpha1.ConditionReasonProviderSuccess)),
+						"Message": Equal("Provider ensured the dns record"),
+					}),
+				))
+				// make sure healthcheck condition is gone
+				g.Expect(dnsRecord.Status.Conditions).ToNot(ContainElements(
+					MatchFields(IgnoreExtras, Fields{
+						"Type": Equal(string(v1alpha1.ConditionTypeHealthy)),
+					}),
+				))
+			}, TestTimeoutLong, time.Second, ctx).Should(Succeed())
+		})
 	})
 
-	Context("On-cluster healthchecks", func() {
-		// TODO test case
+	// multiple controllers against the same host
+	Context("Multiple publishers", func() {
```

Comment:
do you want to add this as part of this PR?

Reply:
Don't have a strong opinion. Removed